test/cpp/api/dataloader.cpp

```diff
@@ -20,6 +21,7 @@
 #include <vector>
 
 using namespace torch::data; // NOLINT
+using namespace c10;
```

Comment:
I'd prefer not to have this using directive here. Let's just write `torch::optional`.